### PR TITLE
core: Allow merge floats

### PR DESCRIPTION
**Pull request by crissebasbol**
Thank you for contributing to LangChain!

- [x] **PR title**: "package: description"
  - Where "package" is whichever of langchain, community, core, experimental, etc. is being modified. Use "docs: ..." for purely docs changes, "templates: ..." for template changes, "infra: ..." for CI changes.
  - Example: "community: add foobar LLM"

- [x]  **Issue:** the issue #16422 to allow google finance and yahoo finance. 

- [x] **Lint and test**: Run `make format`, `make lint` and `make test` from the root of the package(s) you've modified. See contribution guidelines for more: https://python.langchain.com/docs/contributing/

Additional guidelines:
- Make sure optional dependencies are imported within a function.
- Please do not add dependencies to pyproject.toml files (even optional ones) unless they are required for unit tests.
- Most PRs should not touch more than one package.
- Changes should be backwards compatible.
- If you are adding something to community, do not re-import it in langchain.

If no one reviews your PR within a few days, please @-mention one of baskaryan, efriis, eyurtsev, hwchase17.

### CODE REVIEW

File: libs/core/langchain_core/utils/_merge.py [hunk 1]
```diff
@@ -30,6 +30,8 @@ def merge_dicts(left: Dict[str, Any], right: Dict[str, Any]) -> Dict[str, Any]:
             )
         elif isinstance(merged[right_k], str):
             merged[right_k] += right_v
+        elif isinstance(merged[right_k], float):
```
Comment (eyurtsev) [inline libs/core/langchain_core/utils/_merge.py:33]: Why is this change necessary?

**Comment by baskaryan** [comment]
we don't want to merge numerics because there's no generic way to merge them. eg. a latest timestamp should be overwritten, a count should be added, a max should be maxxed, etc. and from just an int/float you don't know which is the correct operation. the best solution is to either drop numerics from a dict before merging or to turn a number into a list of numbers, so that you can append to it and handle combining the numbers outside the merge function.

**Comment by crissebasbol** [comment]
> we don't want to merge numerics because there's no generic way to merge them. eg. a latest timestamp should be overwritten, a count should be added, a max should be maxxed, etc. and from just an int/float you don't know which is the correct operation. the best solution is to either drop numerics from a dict before merging or to turn a number into a list of numbers, so that you can append to it and handle combining the numbers outside the merge function.

But we don't generate the dicts, this is generated automatically by the tools, we have the same problem for Google Finance and yahoo finance using agent executors with vertex as llm, multiple people have the same problem if you see the issues in the repository, some people with float and other people with ints.

**Comment by crissebasbol** [comment]
> numerics from a dict before merging or to turn a number into a list o

I did the change locally and it worked perfect, so why this is not a valid option?

**Comment by crissebasbol** [comment]
> we don't want to merge numerics because there's no generic way to merge them. eg. a latest timestamp should be overwritten, a count should be added, a max should be maxxed, etc. and from just an int/float you don't know which is the correct operation. the best solution is to either drop numerics from a dict before merging or to turn a number into a list of numbers, so that you can append to it and handle combining the numbers outside the merge function.

@baskaryan can you review this again, please?
